impl_unix.go

```diff
@@ -301,7 +312,14 @@ func listenUDP(netw, addr string) (c net.Conn, err error) {
 }
 
 // this is close to the connect() function inside stdlib/net
-func connect(fd int, ra syscall.Sockaddr, deadline time.Time) error {
+func connect(ctx context.Context, fd int, ra syscall.Sockaddr, deadline time.Time) error {
+	// NOTE: We currently only use the context to pass in a deadline.
+	// Canceling an epoll_wait call is hard.
```

Comment:
Yeah, we still don't have a good way to use this purely via cancel. This does however let us pass the contexts timeout down farther, which is helpful